### PR TITLE
Automatic generation docker image on environment variables

### PR DESCRIPTION
**Pull request by szok**
Automatic generation docker image on environment variables

### CODE REVIEW

File: build_files.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+
+def save_dockerfile(env_conf):
+    envs = [
+        "ONBUILD ENV {} {}".format(
+            key, value
+        ) for key, value in env_conf.items()
+    ]
+
+    content = """FROM ubuntu:12.04
```
Comment (mkurek) [inline build_files.py]: maybe this content should be moved to external file (ex. Dockerfile.tmpl) for readability (and then open this file, read it and insert variables into template) - what do you think?

File: build_files.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        f.write(content)
+
+
+def save_build_sh(build_sh_conf):
+    git_clone_lines = []
+    for conf in build_sh_conf:
+        url = 'git clone {}'.format(config.REPOSITORIES_TYPES[conf['type']])
+        git_clone_lines.append(url % conf)
+
+    content = """#!/bin/bash
```
Comment (mkurek) [inline build_files.py]: same situation here

File: build_files.py (hunk 1 above)
Comment (szok) [inline build_files.py]: good idea

**Comment by mkurek** [comment]
are `base/Dockerfile` and `os/build.sh` files necessary anymore? if not, just remove them

File: build_from_github.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
+import json
```
Comment (mkurek) [inline build_from_github.py]: add some comment what this module do

File: build_files.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def main():
+    env_conf = {}
+    build_sh_conf = []
+    for repo in config.REPOSITORIES:
+        fork = "{}_FORK".format(repo['repo_name'].upper())
+        branch = "{}_BRANCH".format(repo['repo_name'].upper())
+        env_conf[fork] = os.environ.get(fork, repo['owner'])
+        env_conf[branch] = 'develop'
```
Comment (mkurek) [inline build_files.py]: why not `os.environ.get(branch, 'develop')`

File: Dockerfile.tmpl [hunk 5]
```diff
@@ -11,14 +11,19 @@ RUN virtualenv /home/ralph/ && /home/ralph/bin/pip install pip==1.5.6 && /home/r
...
 ADD fixtures.py /home/ralph/fixtures.py
 ONBUILD ADD build.sh /home/ralph/build.sh
+ONBUILD RUN chmod 777 /home/ralph/build.sh
 ONBUILD USER root
 ONBUILD RUN chown ralph:ralph -R /home/ralph/
+ONBUILD RUN mkdir /var/run/sshd
+ONBUILD RUN chmod 0755 /var/run/sshd
 ONBUILD USER ralph
+
+{}
```
Comment (kula1922) [inline Dockerfile.tmpl]: ?? what is that?

File: build.sh.tmpl [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+set -e
+
+cd /home/ralph
+. bin/activate
+rm -rf sources
```
Comment (kula1922) [inline build.sh.tmpl]: It is little unsafe. For example i have linked all workspace dir like sources. So, if i run this code now it remove me all my projects. Do it explicit.

```
rm -rf sources/ralph_assets
...
```

the best way is create --force flag and for example if sources/ralph_assets exist and --force is off then current ralph_assets stay as it is but when --roces is on then ralph_assets will remove and download again.

Comment (mkurek) [inline build.sh.tmpl]: Actually we don't need rm anymore - build is running on docker image so there won't be anything to remove

**Comment by vi4m** [comment]
Remeber about specifing minimal docker version.

**Comment by vi4m** [comment]
LGTM

**Comment by mkurek** [comment]
:ok:

**Pull request merged**